src/System.Memory/tests/Span/SequenceCompareTo.string.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.SpanTests
+{
+    public static partial class SpanTests
+    {
+        [Fact]
+        public static void ZeroLengthSequenceCompareTo_String()
+        {
+            string[] a = new string[3];
+
+            Span<string> first = new Span<string>(a, 1, 0);
+            Span<string> second = new Span<string>(a, 2, 0);
+            int result = first.SequenceCompareTo<string>(second);
+            Assert.Equal(0, result);
+        }
+
+        [Fact]
+        public static void SameSpanSequenceCompareTo_String()
+        {
+            string[] a = { "fourth", "fifth", "sixth" };
+            Span<string> span = new Span<string>(a);
+            int result = span.SequenceCompareTo<string>(span);
+            Assert.Equal(0, result);
+        }
+
+        [Fact]
+        public static void SequenceCompareToArrayImplicit_String()
+        {
+            string[] a = { "fourth", "fifth", "sixth" };
+            Span<string> first = new Span<string>(a, 0, 3);
+            int result = first.SequenceCompareTo<string>(a);
+            Assert.Equal(0, result);
+        }
+
+        [Fact]
+        public static void SequenceCompareToArraySegmentImplicit_String()
+        {
+            string[] src = { "first", "second", "third" };
+            string[] dst = { "fifth", "first", "second", "third", "tenth" };
+            var segment = new ArraySegment<string>(dst, 1, 3);
+
+            Span<string> first = new Span<string>(src, 0, 3);
+            int result = first.SequenceCompareTo<string>(segment);
+            Assert.Equal(0, result);
+        }
+
+        [Fact]
+        public static void LengthMismatchSequenceCompareTo_String()
+        {
+            string[] a = { "fourth", "fifth", "sixth" };
+            Span<string> first = new Span<string>(a, 0, 2);
+            Span<string> second = new Span<string>(a, 0, 3);
+            int result = first.SequenceCompareTo<string>(second);
+            Assert.True(result < 0);
+
+            result = second.SequenceCompareTo<string>(first);
+            Assert.True(result > 0);
+
+            // one sequence is empty
+            first = new Span<string>(a, 1, 0);
+
+            result = first.SequenceCompareTo<string>(second);
+            Assert.True(result < 0);
+
+            result = second.SequenceCompareTo<string>(first);
+            Assert.True(result > 0);
+        }
+
+        [Fact]
+        public static void SequenceCompareToWithSingleMismatch_String()
+        {
+            for (int length = 1; length < 32; length++)
+            {
+                for (int mismatchIndex = 0; mismatchIndex < length; mismatchIndex++)
+                {
+                    string[] first = new string[length];
```

Comment:
Nit, it's optional, but we usually use `var` on the left side when the right side is a constructor.

Reply:
Not a problem 😄 My last commit addresses this remark.